### PR TITLE
Fix force-step operations in the debugger

### PR DESCRIPTION
**Pull request by grammati**
Two operations, `:here` and `:out` currently support a `force?`
parameter, which causes the debugger to skip breakpoints in other
instrumented functions until reaching the specified point.

They are bound to uppercase letters, `H` and `O` respectively. Neither
was working previously, because the `force?` parameter was not being
sent (in the case of `:here`), or was being sent incorrectly (in the
case of `:out`).

### CODE REVIEW

File: cider-debug.el [hunk 1]
```diff
@@ -344,6 +344,7 @@ In order to work properly, this mode must be activated by
       (ignore-errors (exit-recursive-edit)))))
 
 (define-key cider--debug-mode-map "h" #'cider-debug-move-here)
+(define-key cider--debug-mode-map "H" #'cider-debug-move-here)
```
Comment (bbatsov) [inline cider-debug.el]: Same command, different keybinding?

File: cider-debug.el [hunk 2]
```diff
@@ -693,9 +696,12 @@ TARGET is inside it.  The returned list is suitable for use in
...
         (goto-char starting-point)))))
 
-(defun cider-debug-move-here ()
+(defun cider-debug-move-here (&optional force)
   "Skip any breakpoints up to point."
-  (interactive)
+  (interactive (list
+                (ignore-errors
+                  (let ((case-fold-search nil))
+                    (string-match "[[:upper:]]" (string last-command-event))))))
```
Comment (bbatsov) [inline cider-debug.el]: What's this supposed to do?

Comment (grammati) [inline cider-debug.el]: I copied this part from `cider-debug-mode-send-reply`, which all other debug commands (other than `:here`) are bound to. The `force` argument is true only if the command was called with an uppercase key.

Comment (bbatsov) [inline cider-debug.el]: Got it. Guess we can extract this code to a small helper function to remove the duplication.

Also we should mention this behaviour in the manual, as I'm pretty sure right now you and @Malabarba are the only people aware this exists. :-)

File: cider-debug.el (hunk 1 above)
Comment (Malabarba) [inline cider-debug.el]: Yes. The command detects whether it was invoked with shift.

Comment (bbatsov) [inline cider-debug.el]: I understood this when I got a bit farther into the code. Might be useful to add a comment here, though.

**Comment by grammati** [comment]
I added a comment and deleted some unnecessary code that I missed the first time.

**Comment by bbatsov** [comment]
@grammati and what about the duplicated code in `(interactive)`?

**Comment by grammati** [comment]
I moved it into a function, `uppercase-command`.

File: cider-debug.el [hunk 3]
```diff
@@ -383,6 +386,12 @@ In order to work properly, this mode must be activated by
      ["List locals" cider-debug-toggle-locals :style toggle :selected cider-debug-display-locals])
     ["Customize" (customize-group 'cider-debug)]))
 
+(defun uppercase-command ()
```
Comment (bbatsov) [inline cider-debug.el]: Rename this to `cider--uppercase-command-p` or something like this. There are no namespace in Elisp and we have to be careful with unprefixed names. `-p` means `?` in Elisp.

Comment (grammati) [inline cider-debug.el]: Done!

**Pull request merged**